src/com/facebook/buck/apple/AppleTest.java

```diff
@@ -303,6 +304,20 @@ public boolean hasTestResultFiles() {
       } else {
         destinationSpecifierArg = defaultDestinationSpecifier;
       }
+
+      Optional<String> snapshotReferenceImagesPath = Optional.empty();
+      if (this.snapshotReferenceImagesPath.isPresent()) {
+        if (this.snapshotReferenceImagesPath.get().isLeft()) {
+          snapshotReferenceImagesPath = Optional.of(
+              getResolver().getAbsolutePath(this.snapshotReferenceImagesPath.get().getLeft())
+                  .toString());
+        } else if (this.snapshotReferenceImagesPath.get().isRight()) {
+          snapshotReferenceImagesPath = Optional.of(
+              getProjectFilesystem().getRootPath().getFileSystem().toString() +
```

Comment:
Calling `toString()` here is incorrect.  You want `getPathForRelativePath()` in `ProjectFilesystem` instead.